joint_state_controller/src/joint_state_controller.cpp

```diff
@@ -121,17 +178,30 @@ JointStateController::update()
   }
 
   joint_state_msg_.header.stamp = lifecycle_node_->get_clock()->now();
-  size_t i = 0;
-  for (auto joint_state_handle : registered_joint_handles_) {
-    joint_state_msg_.position[i] = joint_state_handle->get_position();
-    joint_state_msg_.velocity[i] = joint_state_handle->get_velocity();
-    joint_state_msg_.effort[i] = joint_state_handle->get_effort();
-
-    dynamic_joint_state_msg_.interface_values[i].values[0] = joint_state_handle->get_position();
-    dynamic_joint_state_msg_.interface_values[i].values[1] = joint_state_handle->get_velocity();
-    dynamic_joint_state_msg_.interface_values[i].values[2] = joint_state_handle->get_effort();
 
-    ++i;
+  // update joint state message and dynamic joint state message
+  for (auto joint_index = 0ul; joint_index < joint_names_.size(); ++joint_index) {
+    auto update_value =
+      [](double & msg_data, const std::shared_ptr<hardware_interface::JointHandle> & joint_handle)
+      {
+        if (joint_handle) {
+          msg_data = joint_handle->get_value();
+        }
+      };
+
+    /// @todo somehow don't use hardcoded indexes directly
+    update_value(joint_state_msg_.position[joint_index], joint_handles_[joint_index][0]);
```

Comment:
could you at least make these hardcoded indices named by putting htem in an ALL_CAPS_CONST please?

Reply:
Done.